client/src/modules/cash/dashboard/dashboard.chart.js

```diff
@@ -0,0 +1,309 @@
+   * @param {object} data
+   */
+  function generateGroupChart(data) {
+    var dataset = getCountData(data, 'group', 'name');
+    var ctx = document.getElementById('groupChart').getContext('2d');
+
+    // clear canvas for previous chart rendering
+    if (vm.groupChart) { vm.groupChart.destroy(); }
+
+    vm.groupChart = new Chart(ctx, {
+      type : 'pie',
+      options : chartOptions($translate.instant('DASHBOARD.CASH.DEBTOR_GROUP')),
+      data : {
+        labels : dataset.labels,
+        datasets : [
+          {
+            label : $translate.instant('DASHBOARD.CASH.PAYMENT'),
+            data : dataset.data,
+            backgroundColor : [
+              '#60c6e0', '#0b97b5', '#287181', '#22c36a', '#207647',
+              '#db9b2b', '#634614', '#e91945', '#be1cb4', '#e3e934',
+            ],
+          },
+        ],
+      },
+    });
+  }
+
+  /** chart options */
+  function chartOptions(text) {
+    return {
+      title : {
+        display : true,
+        text : text,
+        fontSize : 20,
+      },
+      legend : {
+        display : true,
+      },
+    };
+  }
+
+  /** get hour data */
+  function getHourData(dataset, field) {
+    var out = { labels : [], data : [] };
+    if (!dataset[field]) { return out; }
+
+    dataset[field].forEach(function (item) {
+      out.labels.push(String(item.hour).concat('H'));
+      out.data.push(item.count);
+    });
+
+    return out;
+  }
+
+  /** get counted data */
+  function getCountData(dataset, field, label) {
+    var out = { labels : [], data : [] };
+    if (!dataset[field]) { return out; }
+
+    dataset[field].forEach(function (item) {
+      out.labels.push(item[label]);
+      out.data.push(item.count);
+    });
+
+    return out;
+  }
+
+  /**
+   * @function uniformize data
+   * @param {object} data - { data: [], labels: [] }
+   * @param {object} reversedData - { data: [], labels: [] }
+   * @param {string} periodGroup - string in ('hour', 'day', 'month')
+   */
+  function uniformize(data, reversedData, periodGroup) {
+    var hours = [
+      '01H', '02H', '03H', '04H', '05H', '06H', '07H', '08H', '09H', '10H', '11H', '12H',
+      '13H', '14H', '15H', '16H', '17H', '18H', '19H', '20H', '21H', '22H', '23H', '24H',
+    ];
+    var dayofweek = [1, 2, 3, 4, 5, 6, 7];
+    var months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    var labels;
+    var period = {
+      hour  : hours,
+      day   : dayofweek,
+      month : months,
+    };
+
+    var base = period[periodGroup || 'hour'];
+    labels = base.concat(data.labels, reversedData.labels);
+    labels = uniqueSortedArray(labels);
+
+    return {
+      dataset : fillArray(labels, data, 'labels'),
+      reversedDataset : fillArray(labels, reversedData, 'labels'),
+    };
+  }
+
+  /** unique array */
+  function uniqueSortedArray(array) {
+    var unique;
+    var i;
+
+    if (!array.length) { return array; }
+
+    array = typeof (array[0]) === 'number' ? array.sort(sortNumber) : array.sort();
```

Comment:
Instead of doing this check, I propose you make a single function - `sortNumberArray()`.  Most code will use `Array.prototype.sort()`, while those that want to sort numbers can use `sortNumberArray()`.

It will simplify the code, and make sure that those functions which need an array of numbers sorted can ask for it.